### PR TITLE
Increase coverage

### PR DESCRIPTION
**Pull request by DavidAKopriva**
Add coverage for valueDictionary methods.

### CODE REVIEW

File: Testing/Tests/ValueClassTests.f90 [hunk 1]
```diff
@@ -205,5 +221,67 @@ SUBROUTINE FTValueClassTests
...
          CALL releaseFTValue(v)
          CALL FTAssert(test = .NOT.ASSOCIATED(v),msg = "Final release deletes object 6")
+         
+         ALLOCATE(v)
+         CALL v % initWithValue(".true.")
+         CALL FTAssert(v % logicalValue(),msg = ".true. converted to logical")
+         CALL releaseFTValue(v)
+         
+         DO j = 1,2 
+         ALLOCATE(v)
```
Comment (andrewwinters5000) [inline Testing/Tests/ValueClassTests.f90:231]:
```suggestion
            ALLOCATE(v)
```

**Review by andrewwinters5000 — CHANGES_REQUESTED** [review]
LGTM! Just a small syntax suggestion.

File: Testing/Tests/ValueClassTests.f90 (hunk 1 above)
Comment (DavidAKopriva) [reply Testing/Tests/ValueClassTests.f90:231]: Done. Will appear in the next commit.

File: Source/FTTesting/Assert.f90 [hunk 2]
```diff
@@ -357,45 +357,56 @@ END SUBROUTINE assertEqualTwoIntegers
...
-             
-             PRINT *, "assertEqualTwoIntegerArrays1D not implemented"
-         END IF 
-         
-      END SUBROUTINE assertEqualTwoIntegerArrays1D
+!      SUBROUTINE assertEqualTwoIntegerArrays1D(expectedValue,actualValue)  
+!
+!TODO: Array assertions are not implemented because a good way to report
+!      where the errors occur needs to be devised. Arrays could get big
+!      and there can be a lot of errors to report.
```
Comment (andrewwinters5000) [inline Source/FTTesting/Assert.f90:364]: In a loop we could then make the appropriate comparison and save the result into a temporary `LOGICAL` variable. On the fly we could then report which position fails (if any) and report the troublesome values.

File: Source/FTTesting/Assert.f90 [hunk 3]
```diff
@@ -357,45 +357,56 @@ END SUBROUTINE assertEqualTwoIntegers
...
+!      END SUBROUTINE assertEqualTwoIntegerArrays2D
 !@mark -
 !
 !//////////////////////////////////////////////////////////////////////// 
 ! 
-      SUBROUTINE assertWithinToleranceTwoReal(expectedValue,actualValue,tol,msg)  
+      SUBROUTINE assertWithinToleranceTwoReal(expectedValue,actualValue,relTol,absTol,msg)  
          IMPLICIT NONE  
-         REAL, INTENT(in)           :: expectedValue,actualValue,tol
+         REAL, INTENT(in)           :: expectedValue,actualValue,relTol
```
Comment (andrewwinters5000) [inline Source/FTTesting/Assert.f90:408]:
```suggestion
         REAL, INTENT(IN)           :: expectedValue,actualValue,relTol
```

File: Source/FTTesting/Assert.f90 [hunk 4]
```diff
@@ -419,59 +430,72 @@ END SUBROUTINE assertWithinToleranceTwoReal
...
 !@mark -
 !
 !//////////////////////////////////////////////////////////////////////// 
 ! 
-      SUBROUTINE assertWithinToleranceTwoDouble(expectedValue,actualValue,tol,msg)  
+      SUBROUTINE assertWithinToleranceTwoDouble(expectedValue,actualValue,relTol, absTol, msg)  
          IMPLICIT NONE  
-         DOUBLE PRECISION, INTENT(in) :: expectedValue,actualValue,tol
-         CHARACTER(LEN=*), OPTIONAL   :: msg
+         DOUBLE PRECISION, INTENT(in)             :: expectedValue,actualValue,relTol
```
Comment (andrewwinters5000) [inline Source/FTTesting/Assert.f90:496]:
```suggestion
         DOUBLE PRECISION, INTENT(IN)             :: expectedValue,actualValue,relTol
```

**Review by andrewwinters5000 — CHANGES_REQUESTED** [review]
Nice work @DavidAKopriva adding all the new tests! I know that this process can be very tiresome. Most of my suggestions are formatting related as everything else already looks good.

File: Source/FTTesting/Assert.f90 (hunk 2 above)
Comment (DavidAKopriva) [reply Source/FTTesting/Assert.f90:364]: The procedure called in the assert returns an array marking where the results differ, so the information is available. The only problem is that the number of those could be huge, so reporting so many positions may be impractical. Perhaps some formatting like Julia does for large arrays where only results near the ends are written and ... is used to indicate skipping over a bunch of them. Some decision as to how big an array is "too being" and needs to be truncated would have to be made.

Comment (andrewwinters5000) [reply Source/FTTesting/Assert.f90:364]: Ah, okay. I see what you mean

Comment (andrewwinters5000) [reply Source/FTTesting/Assert.f90:364]: I opened a corresponding issue #26 to keep track on this.

**Review by andrewwinters5000 — APPROVED** [review]
LGTM!

**Pull request merged**